server/src/main/java/org/opensearch/http/DefaultRestChannel.java

```diff
@@ -147,6 +153,8 @@ public void sendResponse(RestResponse restResponse) {
             addCustomHeaders(httpResponse, restResponse.getHeaders());
             addCustomHeaders(httpResponse, threadContext.getResponseHeaders());
 
+            addCustomHeaders(httpResponse, Map.of(SERVER_VERSION, List.of(SERVER_VERSION_VALUE)));
```

Comment:
You can probably define the map as a constant as well:

```
private static final Map<String, List<String>> SERVER_VERSION_HEADER = Map.of(SERVER_VERSION, List.of(SERVER_VERSION_VALUE));
```

Since this is invoked on every response it is probably worth creating the constant to avoid constructing a new Map and List on every invocation.

Reply:
Added as constant